tests/integration/cmd_storage_test.go

```diff
@@ -23,56 +23,95 @@ var _ = Describe("odo storage command tests", func() {
-			expectedStoreC := []string{"storage", "mystorage", "1Gi", "/opt/app-root/src/storage/"}
-			Expect(helper.GjsonMatcher(valuesStoreC, expectedStoreC)).To(Equal(true))
+			When("listing storage in json", func() {
 
-			actualStorageList := helper.Cmd("odo", "storage", "list", "--context", commonVar.Context, "-o", "json").ShouldPass().Out()
+				actualStorageList := ""
 
-			valuesStoreL := gjson.GetMany(actualStorageList, "kind", "items.0.kind", "items.0.metadata.name", "items.0.spec.size")
-			expectedStoreL := []string{"List", "storage", "mystorage", "1Gi"}
-			Expect(helper.GjsonMatcher(valuesStoreL, expectedStoreL)).To(Equal(true))
+				BeforeEach(func() {
+					actualStorageList = helper.Cmd("odo", "storage", "list", "--context", commonVar.Context, "-o", "json").ShouldPass().Out()
+				})
 
-			helper.Cmd("odo", "storage", "delete", "mystorage", "--context", commonVar.Context, "-f").ShouldPass()
+				It("should list output in json format", func() {
+					valuesStoreL := gjson.GetMany(actualStorageList, "kind", "items.0.kind", "items.0.metadata.name", "items.0.spec.size")
+					expectedStoreL := []string{"List", "storage", "mystorage", "1Gi"}
+					Expect(helper.GjsonMatcher(valuesStoreL, expectedStoreL)).To(Equal(true))
+
+				})
+			})
 		})
+
 	})
 
-	Context("when running storage list command to check state", func() {
-		It("should list storage with correct state", func() {
+	When("component is created and pushed", func() {
 
+		BeforeEach(func() {
 			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
 			helper.Cmd("odo", "component", "create", "--s2i", "nodejs", "nodejs", "--app", "nodeapp", "--project", commonVar.Project, "--context", commonVar.Context).ShouldPass()
 			helper.Cmd("odo", "push", "--context", commonVar.Context).ShouldPass()
+		})
 
-			// create storage, list storage should have state "Not Pushed"
-			helper.Cmd("odo", "storage", "create", "pv1", "--path=/tmp1", "--size=1Gi", "--context", commonVar.Context).ShouldPass()
-			StorageList := helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
-			Expect(StorageList).To(ContainSubstring("Not Pushed"))
-
-			// Push storage, list storage should have state "Pushed"
-			helper.Cmd("odo", "push", "--context", commonVar.Context).ShouldPass()
-			StorageList = helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
-			Expect(StorageList).To(ContainSubstring("Pushed"))
-
-			// Delete storage, list storage should have state "Locally Deleted"
-			helper.Cmd("odo", "storage", "delete", "pv1", "-f", "--context", commonVar.Context).ShouldPass()
-			StorageList = helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
-			Expect(StorageList).To(ContainSubstring("Locally Deleted"))
-
+		When("creating storage", func() {
+			BeforeEach(func() {
+				// create storage, list storage should have state "Not Pushed"
```

Comment:
you can remove this comment, this following It is explicit enough

Reply:
understood, I will remove the comments!

Reply:
Updated!